machine.go

```diff
@@ -660,6 +660,24 @@ func (m *Machine) SetMetadata(ctx context.Context, metadata interface{}) error {
 	return nil
 }
 
+// GetMetadata gets the machine's metadata from MDDS
+func (m *Machine) GetMetadata(ctx context.Context) (map[string]interface{}, error) {
+	resp, err := m.client.GetMmds(ctx)
+	if err != nil {
+		m.logger.Errorf("Getting metadata: %s", err)
+		return nil, err
+	}
+
+	payload, ok := resp.Payload.(map[string]interface{})
```

Comment:
I don't know if I like defaulting to `map[string]interface{}` as `[]interface{}` seems just as feasible. I think we need to think about this a little more, maybe use reflection to unmarshal into a structure may be nice.

```go
func(m *Machine) GetMetadata(ctx context.Context, v interface{}) error {
    // TODO: use reflection here
}
```

How do you feel about just updating the client for now and removing the machine `GetMetadata` and adding an issue asking for a higher level abstraction?

Reply:
I thought about reflection too, I wasn't sure if you guys wanted all that code inside of the sdk. I don't think it'll be too bad since the MMDs only allows strings, objects, and arrays of strings. 

I assume for the reflection we would just unmarshal off of the json struct tags?

Reply:
@cpheps - Yea, using json tags is the way to go, and I think this creates a much better interface for users to be able to have the SDK do the unmarshaling for them.

Reply:
Agreed, I'll go down that path. I also noticed the Patch MMDs was not hooked up to the machine, that seems fairly straight forward and I need that functionality so I'm going to throw that onto this PR.